compliance_checker/cf/cf_1_6.py

```diff
@@ -2172,6 +2172,131 @@ def check_aux_coordinates(self, ds):
             ret_val.append(valid_aux_coords.to_result())
         return ret_val
 
+    # IMPLEMENTATION Section 5 Coordinate Systems and Domain
+    def check_coordinates_attribute_format(self, ds):
+        """
+        Checks that the `coordinates` attribute is a space-separated list of valid variable names,
+        and all listed variables exist in the dataset.
+        """
+        ret_val = []
+
+        geophysical_variables = self._find_geophysical_vars(ds)
+
+        for var_name in geophysical_variables:
+            var = ds.variables[var_name]
+            coord_attr = getattr(var, "coordinates", None)
+
+            if not coord_attr:
+                continue  # No coordinates attribute, nothing to check
+
+            check_coords_attrs_format = TestCtx(
+                BaseCheck.HIGH,
+                self.section_titles["5"],
+            )
+
+            # Check that it is a proper string
+            check_coords_attrs_format.assert_true(
+                isinstance(coord_attr, str),
+                f"The 'coordinates' attribute of variable '{var_name}' must be a string.",
+            )
+
+            # Check for unexpected characters (e.g., commas or brackets)
+            if isinstance(coord_attr, str):
+                has_invalid_chars = any(
+                    char in coord_attr for char in [",", "[", "]", "(", ")"]
```

Comment:
Could use a `set` here to avoid needing to make multiple passes over the string.  Unlikely to have a significant performance impact for such a short string, so it's not necessary.